src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTests.cs

```diff
@@ -17174,5 +17174,135 @@ static unsafe void CheckNaN(float nan)
 ";
             var compilation = CompileAndVerify(source, options: TestOptions.ReleaseExe.WithAllowUnsafe(true), verify: Verification.Skipped, expectedOutput: @"");
         }
+
+        [WorkItem(47887, "https://github.com/dotnet/roslyn/issues/47887")]
+        [Fact]
+        public void ArrayAccessNInt()
```

Comment:
>ArrayAccessNInt [](start = 20, length = 15)

We should also test array access for multidimensional arrays, including those that have rank 1.

Reply:
Can you give an example of a multidimensional array with rank 1?

My understanding is that `Rank` returns `1` for single-dimensional arrays only: https://docs.microsoft.com/en-us/dotnet/api/system.array.rank?view=net-5.0

Reply:
> Can you give an example of a multidimensional array with rank 1?

They cannot be declared in C#, but IL has a syntax for that, for example. I believe, it is enough to include lower bound into the type syntax. Imagine that there is a method that returns an array type like that and in C# we call it and try index into the array. 

---
In reply to: [568003646](https://github.com/dotnet/roslyn/pull/50910#discussion_r568003646) [](ancestors = 568003646)

Reply:
Trying to figure this out. `II.14.2` of ECMA-335 defines the supported array types: 
![image](https://user-images.githubusercontent.com/10487869/106499504-87f71400-6475-11eb-9b15-4ce5a74ab8b7.png)

Given this:
* `int32[]` should be a single-dimensional array starting from zero and with unknown upper bounds, constructed via `newarr`
* `int32[5]` should be a single-dimensional array starting from zero and having an upper bound no more than 5. It isn't clear if these are supported via `newarr` or only via the special constructors
* `int32[1...]` should be a single-dimensional array starting from one and having an unknown upper bounds, it must be constructed via the special constructors
* `int32[1...6]` should be a single-dimensional array starting from one and having an upper bound no more than 6 

Based on this and the comment:
> A constructor that takes twice as many int32 arguments as there are dimensions of the array. These arguments occur in pairs—one pair per dimension—with the first argument of each pair specifying the lower bound for that dimension, and the second argument specifying the total number of elements in that dimension. Note that vectors are not created with this constructor, since a zero lower bound is assumed for vectors. 

One would assume that you would have:
```
    ldc.i4.1
    ldc.i4.5
    newobj instance void int32[1...]::.ctor(int32, int32)
```

and it would construct a single-dimensional array with a lower bound of 1 and an upper bound of `5`. However, it currently throws invalid program

Reply:
ah, nevermind. I just forgot to up `.maxstack` for those methods.

Looks like C# currently supports consuming such arrays (`int32[5]`, `int32[1...]`, and `int32[1...6]`) and recognizes them as `int[*]`: 
![image](https://user-images.githubusercontent.com/10487869/106501236-bbd33900-6477-11eb-86a6-59fb54332f2b.png)

Will get a test up to validate these are handled correctly with `nint`/`nuint` indices

Reply:
> However, it currently throws invalid program

See if you can use an API on System.Array to create an instance.

---
In reply to: [568039545](https://github.com/dotnet/roslyn/pull/50910#discussion_r568039545) [](ancestors = 568039545)